### PR TITLE
Create room

### LINKED ISSUE #48: Support Room Creation

### PR DESCRIPTION
**Pull request by dipoosinubi**
This pull request is for creating a room.

### CODE REVIEW

File: serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt [hunk 1]
```diff
@@ -67,7 +67,26 @@ class MatrixSession(val client: HttpClient, val server: String, val user: String
 
     fun getRoomEvents(id: String) = synchronized(this) { sync_response!!.rooms.join[id]!!.timeline.events }
 
+    fun createRoom(name: String, room_alias_name: String, topic: String): Outcome<String> {
+        try {
+            val result = runBlocking {
+                val creation_confirmation =
+                    client.post<EventIdResponse>("$server/_matrix/client/r0/createRoom?access_token=$access_token") {
```
Comment (lotusronin) [inline serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt]: Probably should either send the transaction id, if it is accepted, or get rid of the transaction id increment+db update down below.

**Review by Limvot — APPROVED** [review]
With LotusRonin's transaction id change, approved

**Review by lotusronin — CHANGES_REQUESTED** [review]
Looks good other than the transaction id.

**Review by lotusronin — APPROVED** [review]
👍

**Review by Limvot — APPROVED** [review]

**Pull request merged**